### ISSUE TITLE
Doxygen support in tooltips

### ISSUE THREAD

**Issue by jansol** [zed-industries/extensions#519]
### Check for existing issues

- [X] Completed

### Describe the feature

Follow-up from zed-industries/zed#5992 as per comments there.

TL;DR: Doc comments a la Doxygen or Javadoc can provide very helpful descriptions e.g. for individual function parameters. These descriptions don't currently get parsed and instead the doc comment is replicated verbatim in the hover tooltip. It would be better if the doc comment's `\param` or `@param` component would be extracted and shown as help for the respective parameter in the function hover tooltip. Ideally an individual parameter's help could even be shown when hovering over that parameter in a function call, in addition to the hover tooltip for the value/variable that is written there.

This appears to be in part a limitation of language servers:
* [clangd](https://github.com/clangd/clangd/issues/529)
* [ccls](https://github.com/MaskRay/ccls/issues/779)

No idea what the status of other languages is.

**Comment by janlarres** [comment]
Starting in version 22 this is handled by clangd itself, you can enable it with this setting in `.clangd`:

```yaml
Documentation:
  CommentFormat: Doxygen
```

See https://clangd.llvm.org/config#documentation for more information.
